### ISSUE TITLE
IE8 with 2.3.0 beta - sizes with calc()

### ISSUE THREAD

**Issue by christopherdarling** [scottjehl/picturefill#450]
On IE8 (Windows XP) picturefill errors out when using sizes="calc(100vw - 2em)", the error given is;

```
Picturefill 2.2.0 (gh-pages branch) - https://raw.githubusercontent.com/scottjehl/picturefill/gh-pages/dist/picturefill.js
Invalid Argument
picturefill.js
Code 0, Line 133, Char 3
```

```
Picturefill 2.3.0 Beta - https://raw.githubusercontent.com/scottjehl/picturefill/2.3/dist/picturefill.js
1st Error; Object doesn't support this property or method
picturefill.js
Code: 0, Line: 602, Char: 3

2nd Error; Invalid Argument
picturefill.js
Code: 0, Line 164, Char 3
```

Changing the sizes value to sizes="100vw" doesn't throw an error on either versions and does display an image, where as with calc() no image is displayed.

**Comment by aFarkas** [comment]
This is due to the fact, that CSS values are not calculated with JS, but handed over to the browser. In conjunction with this neither media conditions nor media queries are supported in IE8.

We currently don't know whether we want to fully fix IE8 issues, partially fix them or add a plugin for IE8 support. In case IE8 support is crucial you can try this version: https://github.com/scottjehl/picturefill/tree/afarkas-pf3-proposal

Note: you need to download the hole repo, ``npm install`` and build it your own with ``grunt``.

**Comment by christopherdarling** [comment]
Thanks for the quick reply Alexander.

I personally don't feel as though IE8 warrants a full polyfill, but at least displaying one image (first, last, or just replace calc()'s with 100vw/%) would be acceptable rather than no image at all.  At the bottom of the readme it mentions that there's currently no known unsupported browsers, it may be worthwhile adding a note about IE8 and being unable to use calc() in the sizes attribute if that's a browser you're wanting to support?

The errors that I mentioned above for 2.3.0 after looking into them look to be down to the placement of the pf.getAllElements function.  The function detectTypeSupport() is being called on line 203 which calls picturefill() onerror or onload of the dummy Image object. I've found that moving the pf.getAllElements() function above detectTypeSupport() resolves the errors appearing that I described in my original ticket, but still no images being loaded.

**Comment by christopherdarling** [comment]
I've added a couple of commits here (https://github.com/christopherdarling/picturefill/tree/issue-450) to demonstrate my thoughts on a couple of things

1. fixing the code placement of getAllElements to stop IE8 erroring out
2. adding a test for calc() (based on Modernizr's test) and falling back on 100vw - this now means that sizes="calc(100vw - 2em)" for browsers that don't support calc() like IE8 will get sizes="100vw"

**Comment by christopherdarling** [comment]
Fixed in 2.3.0 - closing this issue.

Thank you
